### ISSUE TITLE
Project Thumbnail "Upload a File" Button Lacks Focus Indicator

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2439]
As mentioned in the title, and in a previous issue, this specific instance of the "Upload a File" Button has no focus indicator. It can be found on the  General Attributes Page in the Admin Interface for an editable project.

![image](https://user-images.githubusercontent.com/43553368/68515516-25917280-0236-11ea-91f0-f7b803b24c11.png)

This is likely due to its unique style as a sort of merged button with the button to the left of it.

**Comment by dananjohnson** [comment]
@Santoz2 This button lacks a focus indicator because we're actually hiding it using `aria-hidden="true"` and `tabindex="-1"`. The reason for hiding it is that we're providing an `<input type="file" />` for the upload. Our assumption was that we should hide the "Upload" button for screen readers and just announce the `input` so that the readout isn't redundant. Does that seem like a good approach to you? Or should we announce both the `button` and `input? We can update this either way to give a visual indication of focus. But I wanted to confirm that we're approaching the markup correctly first. Thanks!

**Comment by Santoz2** [comment]
@dananjohnson Sorry for the very late response, it has been a very busy quarter! I saw that the aria-hidden attribute was removed. This is definitely a step in the right direction! Now, using tab navigation, a user would land on the button only once, so keyboard navigation is covered. Also, a screen reader can browse the button and read the full text. 

However, because of the multiple buttons, (the overall card, and the upload a file button), it becomes a bit confusing. Hadi has a few reservations about this. Particularly, he is wondering why there are two buttons that do the same thing. I am wondering if it would make sense to make "upload a file" just plain text. To me, the look of the overall button already makes it clear what to click on to upload a file.

Further, now there is the issue that using tab navigation with a screenreader does not read the label of the button. The button as a whole reads "Clickable browse" without context. This is a particular issue in the Description and Images pane, where there are two of these buttons back to back.

![image](https://user-images.githubusercontent.com/43553368/70169728-38099b00-1680-11ea-9184-d9921e13d01c.png)

Using tab to navigate these two would read "Clickable browse, clickable browse", giving no context to either the Background Image or Cover Image buttons. 

Back to this specific project thumbnail issue though, I believe it is separate from the aria-hidden and tabindex fixes used with the other buttons. Notably, the button on the left (default), always has a focus indicator on the page, which notifies visual users that it is their current selection for this option. However, the custom button never shows a focus indicator.

Sorry again for the long and drawn out message, and the delay!

**Comment by dananjohnson** [comment]
@Santoz2 it's my turn to apologize! Thanks for your thorough response. I'm in agreement about the redundant "upload" button. I've created a PR (#2555) that takes care of this issue; it also adds a visual indicator for when the input is focused, and resolves a couple other accessibility issues I noticed along the way.

Thanks again for your ongoing help!

**Comment by sashathomas** [comment]
UW ATX Verification: [#2439](https://github.com/ManifoldScholar/manifold/issues/2439) is closed.
